addons/shortcodes.php

```diff
@@ -59,179 +77,280 @@ function shutdown() {
 	}
 
 	/**
-	 * Callback for the [camptix_attendees] shortcode.
+	 * Routine to preemptively cache the content for all of a site's instances of
+     * the [camptix_attendees] shortcode.
 	 */
-	function shortcode_attendees( $attr ) {
-		global $post, $camptix;
-
-		$attr = shortcode_atts( array(
-			'order' => 'ASC',
-			'orderby' => 'title',
-			'posts_per_page' => 10000,
-			'tickets' => false,
-			'columns' => 3,
-			'questions' => '',
-		), $attr, 'camptix_attendees' );
-
-		$camptix_options = $camptix->get_options();
+	public function cache_all_attendees_shortcodes() {
+        // Get posts containing the `camptix_attendees` shortcode
+		$params = array(
+			'post_type'              => 'page',
+			'post_status'            => 'publish',
+			's'                      => '[camptix_attendees',
+			'posts_per_page'         => 50,
+			'update_post_term_cache' => false,
+			'update_post_meta_cache' => false,
+		);
+		$posts  = get_posts( $params );
 
-		// Enqueue Jetpack's spinner script if available
-		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
-			wp_enqueue_script( 'jquery.spin' );
+		if ( ! $posts ) {
+			return;
 		}
 
-		// For wp.template()
-		wp_enqueue_script( 'wp-util' );
+		$regex = get_shortcode_regex( array( 'camptix_attendees' ) );
 
-		// Lazy load the camptix js.
+		foreach ( $posts as $post ) {
+			$matches = array();
+
+			if ( ! preg_match_all( $regex, $post->post_content, $matches, PREG_SET_ORDER ) ) {
+				continue;
+			}
+
+			foreach ( $matches as $match ) {
+				$attr = shortcode_parse_atts( $match[3] );
+				$attr = $this->sanitize_attendees_atts( $attr );
+
+                $this->get_attendees_shortcode_content( $attr );
+			}
+		}
+	}
+
+	/**
+	 * Callback for the [camptix_attendees] shortcode.
+	 */
+	public function shortcode_attendees( $attr ) {
+		// Required scripts
+		wp_enqueue_script( 'wp-util' ); // For wp.template()
+		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
+			wp_enqueue_script( 'jquery.spin' ); // Enqueue Jetpack's spinner script if available
+		}
 		wp_enqueue_script( 'camptix' );
 
 		// Only print the JS template once.
 		if ( ! has_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) ) ) {
 			add_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) );
 		}
 
-		$start = microtime(true);
+		$attr = $this->sanitize_attendees_atts( $attr );
+
+		/**
+		 *
+		 */
```

Comment:
I think it's a good habit to manually review the output of `git diff` before committing, to catch things like this.